src/telemetry/surveys/activeUseSurvey.ts

```diff
@@ -0,0 +1,193 @@
+
+            tryUpdate(state, lastUseDateKey, getIsoDateString(lastUseDate));
+        }
+
+        //
+        // If the user invoked a command that caused the extension to be activated, the activation would be followed by activation events and then "real use" events.
+        // It would be weird to ask the user "why aren't you using our tools" immediately after the user *just used our tools*. Hence, we wait for a period to allow
+        // those events to flow through, potentially resetting the last use date, before checking whether the survey prompt is needed.
+        //
+
+        let postActivationTimer: NodeJS.Timeout;
+
+        const postActivationTask = new Promise<void>(
+            resolve => {
+                postActivationTimer = setTimeout(
+                    async () => {
+                        try {
+                            const activationTime = activationDate.getTime();
+                            const lastUseTime = lastUseDate.getTime();
+                            const useLimitTime = useLimitInDays * 24 * 60 * 60 * 1000; // day * hour/day * min/hour * sec/min * ms/sec
+                            const inactiveTime = activationTime - lastUseTime;
+
+                            // Has it been more than X number of days since the last "real use" by the user?
+                            if (inactiveTime >= useLimitTime) {
+                                sendSurveyTelemetryEvent('surveys.activeUseSurvey.inactive', { inactiveTimeInMs: inactiveTime.toString() })
+
+                                // Is the user a known candidate (or not)?
+                                let isCandidate = state.get<boolean>(isCandidateKey);
+
+                                // If undecided, run the user through the selection process...
+                                if (isCandidate === undefined) {
+                                    isCandidate = selector();
+
+                                    // ...and update the user as a known candidate (or not).
+                                    await state.update(isCandidateKey, isCandidate);
+                                }
+
+                                // Is the user (assumed to be) a speaker of the survey-chosen language?
+                                // NOTE: The user remains a candidate in case she switches languages.
+                                if (!isChosenLanguage()) {
+                                    isCandidate = false;
+                                }
+
+                                // Is the user still considered a candidate?
+                                if (isCandidate) {
+                                    sendSurveyTelemetryEvent('surveys.activeUseSurvey.eligibleCandidate');
+
+                                    const response = await promptForSurvey();
+
+                                    // Regardless of the response, user is no longer a candidate.
+                                    await state.update(isCandidateKey, false);
```

Comment:
So no Ask Me Later like the NPS one?

Reply:
Correct.  The mechanics of how to track when "later" was was more complicated than the need for that option, so we just went with a simple yes/no.